crates/sui-types/src/deny_list_v2.rs

```diff
@@ -141,37 +145,50 @@ pub fn check_coin_deny_list_v2_during_execution(
             .entry(coin_type.to_canonical_string(false))
             .or_insert_with(BTreeSet::new)
             .insert(owner);
-        num_regulated_transfers += 1;
+        num_non_gas_coin_owners += 1;
     }
-    for (coin_type, owners) in new_coin_owners {
-        let Some(deny_list) = get_per_type_coin_deny_list_v2(&coin_type, object_store) else {
-            continue;
-        };
+    let new_regulated_coin_owners = new_coin_owners
+        .into_iter()
+        .filter_map(|(coin_type, owners)| {
+            let deny_list_config = get_per_type_coin_deny_list_v2(&coin_type, object_store)?;
+            Some((coin_type, (deny_list_config, owners)))
+        })
+        .collect::<BTreeMap<_, _>>();
+    let result =
+        check_new_regulated_coin_owners(new_regulated_coin_owners, cur_epoch, object_store);
+    DenyListResult {
+        result,
+        num_non_gas_coin_owners,
+    }
+}
+
+fn check_new_regulated_coin_owners(
+    new_regulated_coin_owners: BTreeMap<String, (Config, BTreeSet<SuiAddress>)>,
+    cur_epoch: EpochId,
+    object_store: &dyn ObjectStore,
+) -> Result<(), ExecutionError> {
+    for (coin_type, (deny_list, owners)) in new_regulated_coin_owners {
         if check_global_pause(&deny_list, object_store, Some(cur_epoch)) {
-            return (
-                Err(ExecutionError::new(
-                    ExecutionErrorKind::CoinTypeGlobalPause { coin_type },
-                    None,
-                )),
-                num_regulated_transfers,
-            );
+            return Err(ExecutionError::new(
+                ExecutionErrorKind::CoinTypeGlobalPause {
+                    coin_type: coin_type.clone(),
```

Comment:
Is the clone necessary?

Reply:
From a previous version, surprised clippy didn't find it